atoma-service/src/handlers/completions.rs

```diff
@@ -1322,16 +1348,45 @@ pub mod utils {
                 endpoint: endpoint.to_string(),
             });
         }
-        let response = client
-            .post(format!("{}{}", completions_service_url, COMPLETIONS_PATH))
-            .json(&payload)
-            .send()
-            .await;
+        let (result_sender, result_receiver) = tokio::sync::oneshot::channel();
+        state.request_batcher_sender.send(InferenceRequest {
+            url: format!("{}{}", completions_service_url, COMPLETIONS_PATH),
+            payload: payload.clone(),
+            result_sender
+        }).map_err(|e| {
```

Comment:
I suggest we encapsulate this into a handle error method because this is a lot of deduplication of verbose code

Reply:
I am fine with handling the `map_err` deduplication in another PR